[ci-skip] Switch ci

OoLunar opened the pull request:
# Summary
This PR switches from AppVeyor to Github CI. As for why this PR was made, those reasons can be found on the DSharpPlus Discord server. 

# Details
The following workflow files were added:
- [Publish Nightly](https://github.com/OoLunar/DSharpPlus/blob/e8eba18b62548c9a96cb267713ae9e526cffa4fc/.github/workflows/publish_nightly.yml), this publishes pre-releases to our Nuget repo everytime a new commit has been made to the master branch.
- [Publish Release](https://github.com/OoLunar/DSharpPlus/blob/e8eba18b62548c9a96cb267713ae9e526cffa4fc/.github/workflows/publish_release.yml), this publishes official releases to our Nuget repo.
- [Test PR](https://github.com/OoLunar/DSharpPlus/blob/e8eba18b62548c9a96cb267713ae9e526cffa4fc/.github/workflows/test_pr.yml), this builds the PR, ensuring there are no compiler errors.

All workflows support the following:
- Run on Ubuntu
- Should theoretically still use the correct nightly build number
- Use the latest version of the .Net 6.0 SDK
- Supports the `[ci-skip]` tag in both the title and anywhere in the commit message.

After creating a new release, the [Publish Release](https://github.com/OoLunar/DSharpPlus/blob/e8eba18b62548c9a96cb267713ae9e526cffa4fc/.github/workflows/publish_release.yml) workflow should release the packages to Nuget and modify the Github release to add the `*.nupkg` files, just as we had done manually before.

# In the future
- On every commit to the master branch, the [Publish Nightly](https://github.com/OoLunar/DSharpPlus/blob/e8eba18b62548c9a96cb267713ae9e526cffa4fc/.github/workflows/publish_nightly.yml) workflow should be able to modify the #lib-discussion channel topic to specify what the latest nightly is by using the `DSharpPlus.Test` project.

- On every Github Release, the [Publish Release](https://github.com/OoLunar/DSharpPlus/blob/e8eba18b62548c9a96cb267713ae9e526cffa4fc/.github/workflows/publish_release.yml) workflow should be able to modify the #lib-discussion channel topic to specify what the latest stable version is by using the `DSharpPlus.Test` project. Additionally and optionally, it could post an announcement to #announcements with the content's of the Github Release notes.

- The [Test PR](https://github.com/OoLunar/DSharpPlus/blob/e8eba18b62548c9a96cb267713ae9e526cffa4fc/.github/workflows/test_pr.yml) workflow should be able to run the `DSharpPlus.Test` project by utilizing Github secrets.

# Notes
Someone with the correct repo permissions, presumably @Neuheit, should modify the DSharpPlus repo secrets to add a token for Nuget, and a separate Discord token for the `DSharpPlus.Test` project (when we get it correctly setup).

Allows for #1216 to be merged.

OoLunar commented:
I forgot to mention that this depreciates AppVeyor and @Emzi0767's SlimGet feed.

Giggitybyte commented:
The readme should be modified to reflect this change.

OoLunar commented:
Will do shortly

VelvetThePanda commented:
> I forgot to mention that this depreciates AppVeyor and @Emzi0767's SlimGet feed.

Though you're not *wrong*, the word you're probably looking for is *deprecate*(s)

OoLunar commented:
A few changes were made to the README:
- We use Github's action badge since I couldn't find one on https://shields.io
- Updated shield.io badges to get the latest stable and pre-release packages (aka latest stable and latest nightly)
- Updated installation instructions for simplicity

IDoEverything commented on README.md:
```diff
@@ -3,29 +3,33 @@
 You can install the library from following sources:
 
-1. The latest nightly release is available on [SlimGet](https://nuget.emzi0767.com/gallery/packages). These are cutting-edge versions automatically built from the latest commit in the `master` branch in this repository, and as such always contains the latest changes.
+1. The latest nightly release is available on [Nuget](https://www.nuget.org/packages/DSharpPlus/) as a pre-release. These are cutting-edge versions automatically built from the latest commit in the `master` branch in this repository, and as such always contains the latest changes. If you want to use the latest feature on Discord, you should use the nightlies.
```
```suggestion
1. The latest nightly release is available on [Nuget](https://www.nuget.org/packages/DSharpPlus/) as a pre-release. These are cutting-edge versions automatically built from the latest commit in the `master` branch in this repository, and as such always contains the latest changes. If you want to use the latest features on Discord, you should use the nightlies.
```

IDoEverything commented on README.md:
```diff
@@ -3,29 +3,33 @@
 
    Despite the nature of pre-release software, all changes to the library are held under a level of scrutiny; for this library, unstable does not mean bad quality, rather it means that the API can be subject to change without prior notice (to ease rapid iteration) and that consumers of the library should always remain on the latest version available (to immediately get the latest fixes and improv ...
-2. The latest stable release is always available on [NuGet](https://nuget.org/packages/DSharpPlus). This branch is less up-to-date than the nightly versions, but is guaranteed to not receive any breaking API changes without a major version bump.
+2. The latest stable release is always available on [NuGet](https://nuget.org/packages/DSharpPlus). Stable versions are released less often, but is guaranteed to not receive any breaking API changes without a major version bump.
```
```suggestion
2. The latest stable release is always available on [NuGet](https://nuget.org/packages/DSharpPlus). Stable versions are released less often, but are guaranteed to not receive any breaking API changes without a major version bump.
```

IDoEverything reviewed (changes requested):
Grammar
